tower_cli/api.py

```diff
@@ -36,7 +36,23 @@
 TOWER_DATETIME_FMT = r'%Y-%m-%dT%H:%M:%S.%fZ'
 
 
-class TowerTokenAuth(AuthBase):
+class BasicTowerAuth(AuthBase):
+
+    def __new__(cls, username, password, cli_client):
+        if settings.use_token:
+            resp = cli_client._make_request(
+                'OPTIONS', cli_client.prefix + 'authtoken/', [], {}
+            )
+            if not resp.ok:
+                secho(
+                    'use_token is not supported in this version of Tower '
+                    '(the Auth Token API has been replaced with OAuth2.0 support)',
+                    fg='yellow', bold=True
+                )
```

Comment:
I don't feel great about emitting this as `secho`, as opposed to a warning. Example:

https://github.com/ansible/tower-cli/blob/499f4ad39c92f1b619c042ecdf0c559693c7b92d/tower_cli/cli/misc.py#L106-L108

I think the default behavior should still be to show it on every command.